internal/istio/resource/virtualservice.go

```diff
@@ -0,0 +1,110 @@
+package resource
+
+import (
+	"errors"
+	"fmt"
+
+	istioapiv1beta1 "istio.io/api/networking/v1beta1"
+	istioclientapiv1beta1 "istio.io/client-go/pkg/apis/networking/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+)
+
+const (
+	contractVersion = "v1"
+	prefixFormat    = "/%s/%s/event"
+)
+
+func NewVirtualService(targetNamespace string, watcher *v1beta2.Watcher, gateways []*istioclientapiv1beta1.Gateway) (*istioclientapiv1beta1.VirtualService, error) {
+	if targetNamespace == "" {
+		return nil, fmt.Errorf("%w targetNamespace", ErrInvalidArgument)
```

Comment:
Also print the value of targetNamespace (which is empty in this case), or rewrite the error that it shows that the namespace is empty

Reply:
Same for the watcher and watcher.Name.